lib/timers.js

```diff
@@ -147,11 +148,7 @@ function setTimeout(callback, after, arg1, arg2, arg3) {
       args = [arg1, arg2];
       break;
     default:
-      args = [arg1, arg2, arg3];
-      for (i = 5; i < arguments.length; i++) {
-        // Extend array dynamically, makes .apply run much faster in v6.0.0
-        args[i - 2] = arguments[i];
-      }
+      args = [arg1, arg2, ...arg3];
```

Comment:
One thing to consider is that this syntax relies on `Array.prototype[Symbol.iterator]` and `%ArrayIteratorPrototype%.next`, both user-mutable. Maybe it's just a bad idea after all, and we should keep the current implementation.